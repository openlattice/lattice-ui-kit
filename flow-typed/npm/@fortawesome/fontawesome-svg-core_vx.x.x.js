// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
