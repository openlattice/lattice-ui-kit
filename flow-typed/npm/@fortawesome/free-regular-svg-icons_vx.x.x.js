// flow-typed signature: e06b1e176fa03c6cedb9f03bb6cc7872
// flow-typed version: <<STUB>>/@fortawesome/free-regular-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-regular-svg-icons' {
  declare module.exports: any;
}
