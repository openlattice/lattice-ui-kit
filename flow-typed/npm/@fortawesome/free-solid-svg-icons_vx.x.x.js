// flow-typed signature: 64f61153e222e791ad6eca6c60c55baa
// flow-typed version: <<STUB>>/@fortawesome/free-solid-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-solid-svg-icons' {
  declare module.exports: any;
}
