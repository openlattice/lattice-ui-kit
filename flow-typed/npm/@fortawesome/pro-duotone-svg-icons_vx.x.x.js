// flow-typed signature: dc91210f6aed06e504e8c5d05af6775b
// flow-typed version: <<STUB>>/@fortawesome/pro-duotone-svg-icons_v5.10.2/flow_v0.106.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-duotone-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-duotone-svg-icons' {
  declare module.exports: any;
}
