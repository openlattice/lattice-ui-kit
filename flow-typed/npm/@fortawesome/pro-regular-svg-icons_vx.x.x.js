// flow-typed signature: c41877c2f635a0484c823342d488e66e
// flow-typed version: <<STUB>>/@fortawesome/pro-regular-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-regular-svg-icons' {
  declare module.exports: any;
}
