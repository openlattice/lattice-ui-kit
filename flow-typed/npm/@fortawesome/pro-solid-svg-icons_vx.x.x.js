// flow-typed signature: b2f94000db186d565d3eb121a020bf27
// flow-typed version: <<STUB>>/@fortawesome/pro-solid-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-solid-svg-icons' {
  declare module.exports: any;
}
