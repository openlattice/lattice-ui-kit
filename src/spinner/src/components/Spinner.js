import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% {
    transform: rotate(0);
  }

  100% {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.div`
  &::before {
    animation: ${spin} 0.75s linear infinite;
    border: 1px solid #c5d5e5;
    border-radius: 50%;
    border-top-color: #135;
    box-sizing: border-box;
    content: '';
    top: 50%;
    left: 50%;
    width: 50px;
    height: 50px;
    margin-top: -25px; /* half the height */
    margin-left: -25px; /* half the width */
    position: absolute;
  }
`;

export default Spinner;
