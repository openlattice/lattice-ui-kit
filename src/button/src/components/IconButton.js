/*
 * @flow
 */

import React, { useMemo } from 'react';

import clsx from 'clsx';
import styled from 'styled-components';
import { IconButton as MuiIconButton } from '@material-ui/core';

import useButtonStyles, { isCustomColor } from './useButtonStyles';
import { styleName } from './createColorStyles';
import type { ButtonProps } from './types';

import Spinner from '../../../spinner';

const IconSquareWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: auto;

  &::before {
    content: "";
    display: block;
    padding-bottom: 100%; /* padding size is relative to the width */
  }
`;

const SpinnerWrapper = styled.div`
  align-items: center;
  background: white;
  border-radius: 50%;
  display: flex;
  height: 100%;
  justify-content: center;
  position: absolute;
  width: 100%;
`;

const IconButton = ({
  children,
  className,
  color = 'default',
  disabled = false,
  isLoading = false,
  ...rest
} :ButtonProps) => {

  const classes = useButtonStyles();
  const customColor = useMemo(() => isCustomColor(color), [color]);

  // https://medium.com/flow-type/spreads-common-errors-fixes-9701012e9d58
  // [...] you can annotate the object with optional properties and then explicitly set the property afterwards
  const classesObject = {
    [classes[styleName('text', color)]]: customColor,
  };
  if (className) {
    classesObject[className] = className;
  }

  const props = {
    ...rest,
    className: clsx(classesObject),
    color: customColor ? 'default' : color,
    disabled: disabled || isLoading,
  };

  /* eslint-disable react/jsx-props-no-spreading */
  return (
    <MuiIconButton {...props}>
      <IconSquareWrapper>
        {
          isLoading && (
            <SpinnerWrapper>
              <Spinner />
            </SpinnerWrapper>
          )
        }
        {children}
      </IconSquareWrapper>
    </MuiIconButton>
  );
  /* eslint-enable */
};

export default IconButton;
