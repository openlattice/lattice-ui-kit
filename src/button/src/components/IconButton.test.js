/*
 * @flow
 */

import React from 'react';

import _capitalize from 'lodash/capitalize';
import toJson from 'enzyme-to-json';
import { faCode } from '@fortawesome/pro-light-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mount } from 'enzyme';

import IconButton from './IconButton';

import Spinner from '../../../spinner';

const CodeIcon = (
  <FontAwesomeIcon icon={faCode} />
);

const intentColors = ['primary', 'secondary', 'error', 'info', 'success', 'warning'];

describe('IconButton', () => {

  describe('snapshots', () => {

    test('default', () => {
      const wrapper = mount(<IconButton>{CodeIcon}</IconButton>);
      expect(toJson(wrapper)).toMatchSnapshot();
      const button = wrapper.find('button');
      expect(button).toHaveLength(1);
    });

    test('spinner', () => {
      const wrapper = mount(<IconButton isLoading>{CodeIcon}</IconButton>);
      expect(toJson(wrapper)).toMatchSnapshot();
      const button = wrapper.find('button');
      expect(button).toHaveLength(1);
    });

  });

  describe('props', () => {

    test('color', () => {
      intentColors.forEach((color) => {
        const wrapper = mount(<IconButton color={color}>{CodeIcon}</IconButton>);
        const button = wrapper.find('button');
        expect(button.prop('className')).toEqual(expect.stringMatching(`makeStyles-text${_capitalize(color)}`));
      });
      // themeColors.forEach((color) => {
      //   const wrapper = mount(<IconButton color={color}>{CodeIcon}</IconButton>);
      //   const button = wrapper.find('button');
      //   console.log(color);
      //   console.log(button.prop('className'));
      //   expect(button.prop('className')).toEqual(expect.stringMatching(`MuiIconButton-color${_capitalize(color)}`));
      // });
    });

    test('disabled', () => {
      const wrapper = mount(<IconButton disabled>{CodeIcon}</IconButton>);
      const button = wrapper.find('button');
      expect(button.prop('disabled')).toEqual(true);
      expect(button.prop('className')).toEqual(expect.stringMatching('Mui-disabled'));
    });

    test('isLoading', () => {
      const wrapper = mount(<IconButton isLoading>{CodeIcon}</IconButton>);
      const button = wrapper.find('button');
      expect(button.prop('disabled')).toEqual(true);
      expect(button.prop('className')).toEqual(expect.stringMatching('Mui-disabled'));
      expect(wrapper.find(Spinner)).toHaveLength(1);
    });

  });

});
