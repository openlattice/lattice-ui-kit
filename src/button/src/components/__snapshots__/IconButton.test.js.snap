// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton icon with text should match snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  border-radius: 3px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 100ms ease-in-out,border-color 100ms ease-in-out,box-shadow 100ms ease-in-out;
  transition: background-color 100ms ease-in-out,border-color 100ms ease-in-out,box-shadow 100ms ease-in-out;
  white-space: nowrap;
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #8e929b;
}

.c0:hover {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #dcdce7;
  border-color: #dcdce7;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #b6bbc7;
  border-color: #b6bbc7;
  color: #ffffff;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
}

.c0:focus-visible {
  box-shadow: #b6bbc7 0 0 0 2px;
}

.c3 {
  margin: 0 8px 0 0;
}

<IconButton
  icon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f197",
            "M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z",
          ],
          "iconName": "space-shuttle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotate={-45}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
>
  <Button
    disabled={false}
    fontColor=""
    isLoading={false}
    mode="default"
    type="button"
  >
    <StyledButton
      disabled={false}
      fontColor=""
      mode="default"
      type="button"
    >
      <StyledComponent
        disabled={false}
        fontColor=""
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "StyledButton-sc-1etazn9-0",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "border-radius:3px;border-style:solid;border-width:2px;box-sizing:border-box;cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;transition:background-color ",
                "100ms",
                " ease-in-out,border-color ",
                "100ms",
                " ease-in-out,box-shadow ",
                "100ms",
                " ease-in-out;white-space:nowrap;",
                [Function],
                ";:hover{cursor:pointer;text-decoration:none;",
                [Function],
                "};:active{text-decoration:none;",
                [Function],
                "};:disabled{cursor:not-allowed;text-decoration:none;",
                [Function],
                "};:focus-visible{",
                [Function],
                "}",
              ],
            },
            "displayName": "StyledButton",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "StyledButton-sc-1etazn9-0",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        mode="default"
        type="button"
      >
        <button
          className="c0"
          disabled={false}
          mode="default"
          type="button"
        >
          <ContentWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ContentWrapper-sc-36hfub-0",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                    ],
                  },
                  "displayName": "ContentWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ContentWrapper-sc-36hfub-0",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="c1"
              >
                <Content
                  isLoading={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Content-sc-18psqsj-0",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "flex:1 1 auto;opacity:",
                            [Function],
                            ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                          ],
                        },
                        "displayName": "Content",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Content-sc-18psqsj-0",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    isLoading={false}
                  >
                    <span
                      className="c2"
                    >
                      <IconButton__IconMarginRight>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "IconButton__IconMarginRight-g9q5gl-0",
                                "isStatic": true,
                                "lastClassName": "c3",
                                "rules": Array [
                                  "margin:0 8px 0 0;",
                                ],
                              },
                              "displayName": "IconButton__IconMarginRight",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "IconButton__IconMarginRight-g9q5gl-0",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="c3"
                          >
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    640,
                                    512,
                                    Array [],
                                    "f197",
                                    "M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z",
                                  ],
                                  "iconName": "space-shuttle",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotate={-45}
                              rotation={null}
                              size={null}
                              spin={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-space-shuttle fa-w-20 "
                                data-icon="space-shuttle"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                rotate={-45}
                                style={Object {}}
                                viewBox="0 0 640 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </span>
                        </StyledComponent>
                      </IconButton__IconMarginRight>
                      Launch!
                    </span>
                  </StyledComponent>
                </Content>
              </span>
            </StyledComponent>
          </ContentWrapper>
        </button>
      </StyledComponent>
    </StyledButton>
  </Button>
</IconButton>
`;

exports[`IconButton icon without text should match snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  border-radius: 3px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 100ms ease-in-out,border-color 100ms ease-in-out,box-shadow 100ms ease-in-out;
  transition: background-color 100ms ease-in-out,border-color 100ms ease-in-out,box-shadow 100ms ease-in-out;
  white-space: nowrap;
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #8e929b;
}

.c0:hover {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #dcdce7;
  border-color: #dcdce7;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #b6bbc7;
  border-color: #b6bbc7;
  color: #ffffff;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
}

.c0:focus-visible {
  box-shadow: #b6bbc7 0 0 0 2px;
}

<IconButton
  icon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f197",
            "M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z",
          ],
          "iconName": "space-shuttle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotate={-45}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
>
  <Button
    disabled={false}
    fontColor=""
    isLoading={false}
    mode="default"
    type="button"
  >
    <StyledButton
      disabled={false}
      fontColor=""
      mode="default"
      type="button"
    >
      <StyledComponent
        disabled={false}
        fontColor=""
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "StyledButton-sc-1etazn9-0",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "border-radius:3px;border-style:solid;border-width:2px;box-sizing:border-box;cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;transition:background-color ",
                "100ms",
                " ease-in-out,border-color ",
                "100ms",
                " ease-in-out,box-shadow ",
                "100ms",
                " ease-in-out;white-space:nowrap;",
                [Function],
                ";:hover{cursor:pointer;text-decoration:none;",
                [Function],
                "};:active{text-decoration:none;",
                [Function],
                "};:disabled{cursor:not-allowed;text-decoration:none;",
                [Function],
                "};:focus-visible{",
                [Function],
                "}",
              ],
            },
            "displayName": "StyledButton",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "StyledButton-sc-1etazn9-0",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        mode="default"
        type="button"
      >
        <button
          className="c0"
          disabled={false}
          mode="default"
          type="button"
        >
          <ContentWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ContentWrapper-sc-36hfub-0",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                    ],
                  },
                  "displayName": "ContentWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ContentWrapper-sc-36hfub-0",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="c1"
              >
                <Content
                  isLoading={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Content-sc-18psqsj-0",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "flex:1 1 auto;opacity:",
                            [Function],
                            ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                          ],
                        },
                        "displayName": "Content",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Content-sc-18psqsj-0",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    isLoading={false}
                  >
                    <span
                      className="c2"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              640,
                              512,
                              Array [],
                              "f197",
                              "M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z",
                            ],
                            "iconName": "space-shuttle",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotate={-45}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-space-shuttle fa-w-20 "
                          data-icon="space-shuttle"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          rotate={-45}
                          style={Object {}}
                          viewBox="0 0 640 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </span>
                  </StyledComponent>
                </Content>
              </span>
            </StyledComponent>
          </ContentWrapper>
        </button>
      </StyledComponent>
    </StyledButton>
  </Button>
</IconButton>
`;
